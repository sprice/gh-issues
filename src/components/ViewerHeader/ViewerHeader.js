import React from "react"
import PropTypes from "prop-types"

class ViewerHeader extends React.Component {
  render() {
    const { githubUrl } = this.props
    return (
      <header>
        <h2>Github Issue Viewer</h2>
        <h4 className="regular">{githubUrl}</h4>
      </header>
    )
  }
}

ViewerHeader.propTypes = {
  githubUrl: PropTypes.string.isRequired
}

export default ViewerHeader
