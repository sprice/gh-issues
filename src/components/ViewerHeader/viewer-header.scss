header {
  height: 9rem;
  background-color: $pink;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 1rem;
  margin-bottom: 1rem;
  box-shadow: 0px 3px 5px 1px rgba($pink, 0.55);
}

header h2 {
  color: $white;
  font-size: 1.125rem;
  margin-bottom: 0.5rem;
}

header h4 {
  color: $dark-grey;
  word-break: break-all;
}

@media (min-width: $breakpoint-small) {
  header h2 {
    font-size: 1.75rem;
  }
}

@media (min-width: $breakpoint-medium) {
  header {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 5rem;
    padding: 0 4rem;
  }

  header h2 {
    margin-bottom: 0;
  }
}
