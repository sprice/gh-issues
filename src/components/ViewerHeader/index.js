import ViewerHeader from "./ViewerHeader"

export default ViewerHeader
