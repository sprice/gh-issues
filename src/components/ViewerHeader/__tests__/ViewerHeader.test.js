import React from "react"
import { create } from "react-test-renderer"
import ViewerHeader from "../ViewerHeader"

test("Snapshop of ViewerHeader", () => {
  const c = create(
    <ViewerHeader githubUrl="https://github.com/ethereum/solidity" />
  )
  expect(c.toJSON()).toMatchSnapshot()
})
