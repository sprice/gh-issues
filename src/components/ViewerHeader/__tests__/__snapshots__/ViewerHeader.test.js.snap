// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshop of ViewerHeader 1`] = `
<header>
  <h2>
    Github Issue Viewer
  </h2>
  <h4
    className="regular"
  >
    https://github.com/ethereum/solidity
  </h4>
</header>
`;
